*** kernel/common/ezurloperator.php	2005-04-06 04:19:45.000000000 -0700
--- kernel/common/ezurloperator.php.apache2	2005-05-23 16:47:30.000000000 -0700
***************
*** 230,239 ****
--- 230,248 ----
                      {
                          if ( strlen( $url ) > 0 && $url[0] == '#' )
                          {
+ 			    // Examples: Apache/1.3.14 (Win32) or Apache/2.0.40 (Red Hat Linux)
+ 			    // if( strstr( $_ENV["SERVER_SOFTWARE"], "Apache/1.3" ) != "" )
+ 			    if( strstr( $_ENV["SERVER_SOFTWARE"], "Apache/1" ) != "" )	
+ 	                    {
  			        $url = htmlspecialchars( $url );
  			    }		
       			    else 
  			    {	
+                                 $url = htmlspecialchars( $url ) . "/";
+ 			    }
+                         }
+                         else
+                         {
                              if ( strlen( $url ) == 0 )
                              {
                                  $url = '/';
